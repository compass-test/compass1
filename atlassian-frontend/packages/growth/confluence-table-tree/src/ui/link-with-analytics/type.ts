import { ComponentPropsWithoutRef } from 'react';

// https://react-typescript-cheatsheet.netlify.app/docs/advanced/patterns_by_usecase/#type-narrowing-based-on-props
export type LinkPropsWithoutRef = ComponentPropsWithoutRef<'a'>;
