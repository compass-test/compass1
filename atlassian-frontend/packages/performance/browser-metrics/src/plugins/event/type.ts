import { mapEventType } from '../../helper/map-event-type';
import { PerformanceEventConfig } from '../../types';

export const eventType = (config: PerformanceEventConfig) => {
  return {
    'event:type': mapEventType(config.type),
  };
};
