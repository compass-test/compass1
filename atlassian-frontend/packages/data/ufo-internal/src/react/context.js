import React, {
  Component,
  createContext,
  useContext,
  useEffect,
  useRef,
} from 'react';

import { v4 as uuidv4 } from 'uuid';

import {
  ConcurrentExperience,
  UFOExperience,
} from '@atlassian/ufo-experimental';
import { ufolog } from '@atlassian/ufo-experimental/logger';

export const getActions = () => {
  return {
    actions: {},
  };
};

export const UFOContext = createContext(null);

type ErrorBoundaryProps = {
  metric: UFOExperience,
};

class UFOErrorBoundary extends Component<ErrorBoundaryProps> {
  componentDidCatch(/*error, errorInfo*/) {
    this.props.metric.failure();
  }

  render() {
    return this.props.children;
  }
}

type ContextProps = {
  metric: UFOExperience,
  start?: boolean,
};

export const UFOContextComponent = (props: ContextProps) => {
  const context = useContext(UFOContext);

  const hasBeenCalled = useRef(false);
  const firstRun = hasBeenCalled.current === false;

  if (hasBeenCalled.current === false) {
    hasBeenCalled.current = uuidv4();
  }

  //console.log('context', props.metric.id, context);

  const metric =
    props.metric instanceof ConcurrentExperience
      ? props.metric.getInstance(hasBeenCalled.current)
      : props.metric;

  if (context) {
    //metric.setParent(context);
  }
  if ((props.start === undefined || props.start !== false) && firstRun) {
    ufolog('START', props.metric.id, hasBeenCalled.current);
    metric.start();
  }

  useEffect(() => {
    return () => {
      props.metric.release?.(hasBeenCalled.current);
    };
  }, [props.metric]);

  return (
    <UFOContext.Provider value={metric}>
      <UFOErrorBoundary metric={metric}>
        <div data-uuid={`${props.metric.id} ${hasBeenCalled.current}`}>
          {props.children}
        </div>
      </UFOErrorBoundary>
    </UFOContext.Provider>
  );
};
