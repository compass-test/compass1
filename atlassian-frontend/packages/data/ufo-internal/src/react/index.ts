export { useUFOComponentExperience } from './use-component-experience';
export { ExperienceSuccess } from './experience-success';
