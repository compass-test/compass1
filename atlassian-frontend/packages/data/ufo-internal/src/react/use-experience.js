// PLACEHOLDER for concurrent experiences for components
export const useExperience = experience => {
  return experience;
};
