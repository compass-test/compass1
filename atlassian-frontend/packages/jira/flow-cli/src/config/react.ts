export default {
  Component: 'React$Component',
  'React.Component': 'React$Component',
  RefObject: 'React$Ref',
  'React.RefObject': 'React$Ref',
  Ref: 'React$Ref',
  'React.Ref': 'React$Ref',
  ReactNode: 'React$Node',
  'React.ReactNode': 'React$Node',
  ElementType: 'React$ElementType',
  'React.ElementType': 'React$ElementType',
  ReactElement: 'React$Element',
  'React.ReactElement': 'React$Element',
  SFC: 'React$ComponentType',
  'React.SFC': 'React$ComponentType',
  PureComponent: 'React$Component',
  'React.PureComponent': 'React$Component',
  ComponentClass: 'React$ComponentType',
  'React.ComponentClass': 'React$ComponentType',
  ComponentType: 'React$ComponentType',
  'React.ComponentType': 'React$ComponentType',
  StatelessComponent: 'React$ComponentType',
  'React.StatelessComponent': 'React$ComponentType',
  MouseEvent: 'SyntheticMouseEvent',
  'React.MouseEvent': 'SyntheticMouseEvent',
  FocusEvent: 'SyntheticFocusEvent',
  'React.FocusEvent': 'SyntheticFocusEvent',
  KeyboardEvent: 'SyntheticKeyboardEvent',
  'React.KeyboardEvent': 'SyntheticKeyboardEvent',
  ReactPortal: 'React$Portal',
  'React.ReactPortal': 'React$Portal',
  ReactChild: 'React$Node',
  'React.ReactChild': 'React$Node',
  FC: 'React$ComponentType',
  'React.FC': 'React$ComponentType',
  Context: 'React$Context',
  'React.Context': 'React$Context',

  MouseEventHandler: '(event: SyntheticMouseEvent<any>) => void',
  'MouseEventHandler<HTMLElement>':
    '(event: SyntheticMouseEvent<HTMLElement>) => void',
  'MouseEventHandler<HTMLDivElement>':
    '(event: SyntheticMouseEvent<HTMLDivElement>) => void',
  'MouseEventHandler<HTMLButtonElement>':
    '(event: SyntheticMouseEvent<HTMLButtonElement>) => void',
  FocusEventHandler: '(event: SyntheticFocusEvent<any>) => void',
  'FocusEventHandler<HTMLElement>':
    '(event: SyntheticFocusEvent<HTMLElement>) => void',
  'FocusEventHandler<HTMLInputElement>':
    '(event: SyntheticFocusEvent<HTMLInputElement>) => void',
  'FocusEventHandler<HTMLTextAreaElement>':
    '(event: SyntheticFocusEvent<HTMLTextAreaElement>) => void',
  'FocusEventHandler<HTMLDivElement>':
    '(event: SyntheticFocusEvent<HTMLDivElement>) => void',
  'FocusEventHandler<HTMLButtonElement>':
    '(event: SyntheticFocusEvent<HTMLButtonElement>) => void',
  'React.MouseEventHandler': '(event: SyntheticFocusEvent<any>) => void',
  'React.MouseEventHandler<HTMLElement>':
    '(event: SyntheticMouseEvent<HTMLElement>) => void',
  'React.MouseEventHandler<HTMLDivElement>':
    '(event: SyntheticMouseEvent<HTMLDivElement>) => void',
  'React.MouseEventHandler<HTMLButtonElement>':
    '(event: SyntheticMouseEvent<HTMLButtonElement>) => void',
  'React.FocusEventHandler': '(event: SyntheticFocusEvent<any>) => void',
  'React.FocusEventHandler<HTMLElement>':
    '(event: SyntheticFocusEvent<HTMLElement>) => void',
  'React.FocusEventHandler<HTMLInputElement>':
    '(event: SyntheticFocusEvent<HTMLInputElement>) => void',
  'React.FocusEventHandler<HTMLTextAreaElement>':
    '(event: SyntheticFocusEvent<HTMLTextAreaElement>) => void',
  'React.FocusEventHandler<HTMLDivElement>':
    '(event: SyntheticFocusEvent<HTMLDivElement>) => void',
  'React.FocusEventHandler<HTMLButtonElement>':
    '(event: SyntheticFocusEvent<HTMLButtonElement>) => void',

  'PropTypes.Requireable<any>': 'any',
};
