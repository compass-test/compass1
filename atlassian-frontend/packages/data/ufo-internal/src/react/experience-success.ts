import { useLayoutEffect } from 'react';

import { UFOExperience } from '@atlassian/ufo-experimental';

type Props = { experience: UFOExperience };

/**
 * A React component that succeeds the desired experience when mounted. Uses useLayoutEffect to ensure accuracy of success measurement
 * Use this instead of a straight call to `experience.success()` if your success point is during a render.
 * @param experience
 * @constructor
 */
export const ExperienceSuccess = ({ experience }: Props) => {
  useLayoutEffect(() => {
    experience.success();
  }, [experience]);
  return null;
};
