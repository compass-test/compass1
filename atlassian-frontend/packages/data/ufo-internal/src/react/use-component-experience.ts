import { useEffect, useRef } from 'react';

import { UFOExperience } from '@atlassian/ufo-experimental';

const useConstructor = (callback: () => void) => {
  const hasBeenFired = useRef(false);
  if (!hasBeenFired.current) {
    callback();
    hasBeenFired.current = true;
  }
  return null;
};

/**
 * A hook to start an experience and to auto abort the experience when the parent component is unmounted.
 * Use this instead of a direct call to `experience.start`. If you need to restart the experience
 * simply trigger an unmount and remount of the parent component.
 * @param experience the experience to start and abort
 */
export const useUFOComponentExperience = (experience: UFOExperience) => {
  useEffect(() => {
    return () => {
      experience.abort();
    };
    // we want this cleanup to only happen on unmount so this is a legit use of empty array
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  useConstructor(() => experience.start());
  return null;
};
