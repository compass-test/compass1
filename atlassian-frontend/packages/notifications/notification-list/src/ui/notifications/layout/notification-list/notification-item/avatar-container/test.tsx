import React from 'react';

import { render } from '@testing-library/react';

import type { Actor } from './types';

import AvatarContainer from './index';

describe.skip('AvatarContainer', () => {
  test('renders an empty avatar if no actors are supplied', () => {
    const actors: Actor[] = [];
    const { container } = render(<AvatarContainer actors={actors} />);
    expect(container).toMatchSnapshot();
  });
  test('renders if first actor exists', () => {
    const actors: Actor[] = [
      {
        displayName: 'Person',
        ari: 'ASDFADSF/ASDFASDF',
        avatarUrl: 'http://gotoavatar.com/avatar.jpg',
      },
      {
        displayName: 'Person',
        ari: 'ASDFADSF/ASDFASDF',
        avatarUrl: 'http://gotoavatar.com/avatar.jpg',
      },
    ];
    const { container } = render(<AvatarContainer actors={actors} />);
    expect(container).toMatchSnapshot();
  });
  test('renders if first actor exists even when there is no ARI', () => {
    const actors: Actor[] = [
      {
        displayName: 'Person',
        avatarUrl: 'http://gotoavatar.com/avatar.jpg',
      },
      {
        displayName: 'Person',
        ari: 'ASDFADSF/ASDFASDF',
        avatarUrl: 'http://gotoavatar.com/avatar.jpg',
      },
    ];
    const { container } = render(<AvatarContainer actors={actors} />);
    expect(container).toMatchSnapshot();
  });
});
