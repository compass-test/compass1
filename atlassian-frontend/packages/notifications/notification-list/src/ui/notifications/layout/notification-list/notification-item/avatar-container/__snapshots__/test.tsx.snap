// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarContainer renders an empty avatar if no actors are supplied 1`] = `
.emotion-3 {
  margin-right: 8px;
}

.emotion-2 {
  height: 40px;
  width: 40px;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  background-color: #FFFFFF;
  border-radius: 50%;
  box-sizing: content-box;
  cursor: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  outline: none;
  overflow: hidden;
  position: static;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -webkit-transition: -webkit-transform 200ms,opacity 200ms;
  -webkit-transition: transform 200ms,opacity 200ms;
  transition: transform 200ms,opacity 200ms;
  box-shadow: 0 0 0 2px #FFFFFF;
  border: none;
  margin: 2px;
  padding: 0;
  font-size: inherit;
  font-family: inherit;
}

.emotion-2::-moz-focus-inner {
  border: 0;
  margin: 0;
  padding: 0;
}

.emotion-2::after {
  background-color: transparent;
  bottom: 0px;
  border-radius: 50%;
  content: ' ';
  left: 0px;
  opacity: 0;
  pointer-events: none;
  position: absolute;
  right: 0px;
  top: 0px;
  -webkit-transition: opacity 200ms;
  transition: opacity 200ms;
  width: 100%;
}

.emotion-1 {
  background-color: #8993A4;
  width: 100%;
  height: 100%;
  display: block;
}

.emotion-1 svg {
  height: 40px;
  width: 40px;
}

.emotion-0 {
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  line-height: 1;
}

.emotion-0 > svg {
  max-height: 100%;
  max-width: 100%;
  vertical-align: bottom;
  overflow: hidden;
  pointer-events: none;
  color: #FFFFFF;
  fill: #8993A4;
}

.emotion-0 > svg stop {
  stop-color: currentColor;
}

<div>
  <div
    class="emotion-3"
  >
    <div
      style="display: inline-block; position: relative; outline: 0;"
    >
      <span
        class="emotion-2"
      >
        <span
          class="emotion-1"
        >
          <span
            aria-hidden="true"
            class="emotion-0"
            role="presentation"
          >
            <svg
              height="24"
              role="presentation"
              viewBox="0 0 24 24"
              width="24"
            >
              <g
                fill="currentColor"
                fill-rule="evenodd"
              >
                <path
                  d="M6 14c0-1.105.902-2 2.009-2h7.982c1.11 0 2.009.894 2.009 2.006v4.44c0 3.405-12 3.405-12 0V14z"
                />
                <circle
                  cx="12"
                  cy="7"
                  r="4"
                />
              </g>
            </svg>
          </span>
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`AvatarContainer renders if first actor exists 1`] = `
.emotion-3 {
  margin-right: 8px;
}

.emotion-2 {
  height: 40px;
  width: 40px;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  background-color: #FFFFFF;
  border-radius: 50%;
  box-sizing: content-box;
  cursor: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  outline: none;
  overflow: hidden;
  position: static;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -webkit-transition: -webkit-transform 200ms,opacity 200ms;
  -webkit-transition: transform 200ms,opacity 200ms;
  transition: transform 200ms,opacity 200ms;
  box-shadow: 0 0 0 2px #FFFFFF;
  border: none;
  margin: 2px;
  padding: 0;
  font-size: inherit;
  font-family: inherit;
}

.emotion-2::-moz-focus-inner {
  border: 0;
  margin: 0;
  padding: 0;
}

.emotion-2::after {
  background-color: transparent;
  bottom: 0px;
  border-radius: 50%;
  content: ' ';
  left: 0px;
  opacity: 0;
  pointer-events: none;
  position: absolute;
  right: 0px;
  top: 0px;
  -webkit-transition: opacity 200ms;
  transition: opacity 200ms;
  width: 100%;
}

.emotion-1 {
  background-color: #8993A4;
  width: 100%;
  height: 100%;
  display: block;
}

.emotion-1 svg {
  height: 40px;
  width: 40px;
}

.emotion-0 {
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  line-height: 1;
}

.emotion-0 > svg {
  max-height: 100%;
  max-width: 100%;
  vertical-align: bottom;
  overflow: hidden;
  pointer-events: none;
  color: #FFFFFF;
  fill: #8993A4;
}

.emotion-0 > svg stop {
  stop-color: currentColor;
}

<div>
  <div
    class="emotion-3"
  >
    <span
      aria-expanded="false"
      aria-haspopup="true"
    >
      <div
        style="display: inline-block; position: relative; outline: 0;"
      >
        <span
          class="emotion-2"
        >
          <span
            class="emotion-1"
          >
            <span
              aria-label="Person"
              class="emotion-0"
              role="img"
            >
              <svg
                height="24"
                role="presentation"
                viewBox="0 0 24 24"
                width="24"
              >
                <g
                  fill="currentColor"
                  fill-rule="evenodd"
                >
                  <path
                    d="M6 14c0-1.105.902-2 2.009-2h7.982c1.11 0 2.009.894 2.009 2.006v4.44c0 3.405-12 3.405-12 0V14z"
                  />
                  <circle
                    cx="12"
                    cy="7"
                    r="4"
                  />
                </g>
              </svg>
            </span>
          </span>
        </span>
      </div>
    </span>
  </div>
</div>
`;

exports[`AvatarContainer renders if first actor exists even when there is no ARI 1`] = `
.emotion-3 {
  margin-right: 8px;
}

.emotion-2 {
  height: 40px;
  width: 40px;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  background-color: #FFFFFF;
  border-radius: 50%;
  box-sizing: content-box;
  cursor: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  outline: none;
  overflow: hidden;
  position: static;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  -webkit-transition: -webkit-transform 200ms,opacity 200ms;
  -webkit-transition: transform 200ms,opacity 200ms;
  transition: transform 200ms,opacity 200ms;
  box-shadow: 0 0 0 2px #FFFFFF;
  border: none;
  margin: 2px;
  padding: 0;
  font-size: inherit;
  font-family: inherit;
}

.emotion-2::-moz-focus-inner {
  border: 0;
  margin: 0;
  padding: 0;
}

.emotion-2::after {
  background-color: transparent;
  bottom: 0px;
  border-radius: 50%;
  content: ' ';
  left: 0px;
  opacity: 0;
  pointer-events: none;
  position: absolute;
  right: 0px;
  top: 0px;
  -webkit-transition: opacity 200ms;
  transition: opacity 200ms;
  width: 100%;
}

.emotion-1 {
  background-color: #8993A4;
  width: 100%;
  height: 100%;
  display: block;
}

.emotion-1 svg {
  height: 40px;
  width: 40px;
}

.emotion-0 {
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  line-height: 1;
}

.emotion-0 > svg {
  max-height: 100%;
  max-width: 100%;
  vertical-align: bottom;
  overflow: hidden;
  pointer-events: none;
  color: #FFFFFF;
  fill: #8993A4;
}

.emotion-0 > svg stop {
  stop-color: currentColor;
}

<div>
  <div
    class="emotion-3"
  >
    <span
      aria-expanded="false"
      aria-haspopup="true"
    >
      <div
        style="display: inline-block; position: relative; outline: 0;"
      >
        <span
          class="emotion-2"
        >
          <span
            class="emotion-1"
          >
            <span
              aria-label="Person"
              class="emotion-0"
              role="img"
            >
              <svg
                height="24"
                role="presentation"
                viewBox="0 0 24 24"
                width="24"
              >
                <g
                  fill="currentColor"
                  fill-rule="evenodd"
                >
                  <path
                    d="M6 14c0-1.105.902-2 2.009-2h7.982c1.11 0 2.009.894 2.009 2.006v4.44c0 3.405-12 3.405-12 0V14z"
                  />
                  <circle
                    cx="12"
                    cy="7"
                    r="4"
                  />
                </g>
              </svg>
            </span>
          </span>
        </span>
      </div>
    </span>
  </div>
</div>
`;
